mm2src/coins/lp_coins.rs

```diff
@@ -2168,18 +2183,73 @@ impl CoinsContext {
         tokens: Vec<MmCoinEnum>,
     ) -> Result<(), MmError<PlatformIsAlreadyActivatedErr>> {
         let mut coins = self.coins.lock().await;
+        let mut platform_coin_tokens = self.platform_coin_tokens.lock();
+
         if coins.contains_key(platform.ticker()) {
             return MmError::err(PlatformIsAlreadyActivatedErr {
                 ticker: platform.ticker().into(),
             });
         }
 
+        let platform_ticker = platform.ticker().to_string();
         coins.insert(platform.ticker().into(), platform);
 
         // Tokens can't be activated without platform coin so we can safely insert them without checking prior existence
         for token in tokens {
-            coins.insert(token.ticker().into(), token);
+            coins.insert(token.ticker().into(), token.clone());
+            platform_coin_tokens
+                .entry(platform_ticker.clone())
+                .or_default()
+                .insert(0, token.ticker().to_string());
         }
+
+        Ok(())
+    }
+
+    /// Get enabled coins to disable.
+    pub async fn get_tokens_to_disable(&self, ticker: &str) -> Vec<String> {
+        let coins = self.platform_coin_tokens.lock();
+        coins.get(ticker).cloned().unwrap_or_default()
+    }
+
+    pub async fn remove_coin(&self, ctx: &MmArc, ticker: &str) -> Result<(), String> {
+        let coin = match lp_coinfind(ctx, ticker).await {
+            Ok(Some(coin)) => coin,
+            Ok(None) => return ERR!("No such coin: {}", ticker),
+            Err(err) => return ERR!("!lp_coinfind({}): ", err),
+        };
+        let coins_ctx = try_s!(CoinsContext::from_ctx(ctx));
+        let platform_ticker = coin.platform_ticker();
+
+        let mut coins_storage = try_s!(coins_ctx.coins.try_lock().ok_or("coins mutex lock err"));
+        let mut platform_tokens_storage = try_s!(coins_ctx
+            .platform_coin_tokens
+            .try_lock()
+            .ok_or("platform_coin_tokens mutex lock err"));
+
+        // Check if ticker is a platform coin and remove from it platform's token list
+        if ticker == platform_ticker && platform_tokens_storage.get_mut(ticker).is_some() {
+            if let Err(err) = coin.on_token_deactivated(ticker) {
+                log!("Platform Tokens Error: {err}")
+            };
+            platform_tokens_storage.remove(ticker);
+        };
+
+        // Check if coin platform_ticker is in platform_tokens and remove it from token list
+        if let Some(tokens) = platform_tokens_storage.get_mut(platform_ticker) {
+            tokens.retain(|t| t.as_str() != ticker);
+        };
```

Comment:
I have two notes regarding this:
1. `on_token_deactivated(ticker)` should be invoked for a platform coin and if only `ticker != platform_ticker`
2. I'm standing for removing tokens right in this method if `ticker == platform_ticker`. So if we figured out that `ticker = platform_ticker`, then (pseudo-code):
```rust
let tokens_to_remove = platform_tokens_storage.remove(ticker);
for token in tokens_to_remove {
  let token = coins.remove(token);
  token.on_disabled();
}
```

Reply:
make sense. done